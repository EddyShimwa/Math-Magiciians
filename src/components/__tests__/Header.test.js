import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../interface/header';

describe('Use snapshots to confirm the stability of the Header component', () => {
  test('should match the snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should include the logo', () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    expect(
      screen.getByRole('heading', { name: 'Math Magicians', exact: false }),
    ).toBeInTheDocument();
  });

  test('should highlight the active link', () => {
    const { getByRole } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    const homeLink = getByRole('link', { name: 'Home' });
    expect(homeLink).toHaveClass('active');
    const quoteLink = getByRole('link', { name: 'Quote' });
    expect(quoteLink).not.toHaveClass('active');
  });
});
