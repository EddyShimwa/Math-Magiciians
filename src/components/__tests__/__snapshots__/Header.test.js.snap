// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use snapshots to confirm the stability of the Header component should match the snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="nav_bar"
  >
    <div
      className="nav_title"
    >
      <h1>
        <a
          href="/"
          onClick={[Function]}
        >
          Math Magicians
        </a>
      </h1>
    </div>
    <ul
      className="nav_items"
    >
      <li
        className="nav_item"
      >
        <a
          aria-current="page"
          className="nav_link active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav_link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="nav_link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
